src/Controls/src/Core/View/View.cs

```diff
@@ -132,19 +132,22 @@ void RemoveItems(IEnumerable<IElementDefinition> elements)
 						List<IElementDefinition> remove = new List<IElementDefinition>();
 						List<IElementDefinition> add = new List<IElementDefinition>();
 
-						foreach (IElementDefinition item in _gestureRecognizers.OfType<IElementDefinition>())
+						var gestureRecognizers = _gestureRecognizers.OfType<IElementDefinition>().ToHashSet();
+						var compositeGestureRecognizers = GestureController.CompositeGestureRecognizers.OfType<IElementDefinition>().ToList();
```

Comment:
The issue here is it creates two intermediate collections: `HashSet` and `List`. Without a benchmark, it's hard to know if the extra allocations are offset by `HashSet.Contains()`. In a `View` with 1 recognizer (probably most common case), this seems like the performance would be worse?

Looking at the original code, it seems like we'd be better off:

* Change the parameters of `AddItems`/`RemoveItems` so `OfType<IElementDefinition>()` is not needed
* Remove other System.Linq usage, if any
* See if we can avoid creating:

```csharp
List<IElementDefinition> remove = new List<IElementDefinition>();
List<IElementDefinition> add = new List<IElementDefinition>();
```

If you want to continue to iterate on this one, add a BenchmarkDotNet case. Thanks!

Reply:
Done as requested.

Reply:
The three benchmarks involving just one gesture recognizer were actually faster and allocated less.